repo.go

```diff
@@ -76,11 +76,11 @@ func (r *Repo) Init(consistentSnapshot bool) error {
 	}
 	root := data.NewRoot()
 	root.ConsistentSnapshot = consistentSnapshot
-	err = r.setMeta("root.json", root)
-	if err != nil {
-		return err
-	}
-	_, err = r.GenKey("root")
+	return r.setMeta("root.json", root)
```

Comment:
setMeta can't be recovered from internally to the repo code. The called might be able to do something about it as it would typically mean the storage backend is unavailable, or keys have not yet been generated for the role.

On the return question, it seems to be idiomatic go and generally accepted.